util/src/main/java/tc/oc/pgm/util/chunk/NullChunkGenerator.java

```diff
@@ -1,18 +1,53 @@
 package tc.oc.pgm.util.chunk;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Random;
+import org.bukkit.Location;
 import org.bukkit.World;
+import org.bukkit.block.Biome;
+import org.bukkit.generator.BlockPopulator;
 import org.bukkit.generator.ChunkGenerator;
 
 /** A chunk generator that creates empty chunks. */
 public class NullChunkGenerator extends ChunkGenerator {
+  // TODO: PLATFORM DEPENDANT? it could fail in newer versions, but may be fine
   public static final NullChunkGenerator INSTANCE = new NullChunkGenerator();
   private static final byte[] CHUNK = new byte[0];
 
   private NullChunkGenerator() {}
 
-  @Override
   public byte[] generate(final World world, final Random random, final int x, final int z) {
     return CHUNK;
   }
+
+  @Override
+  public List<BlockPopulator> getDefaultPopulators(World world) {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public ChunkData generateChunkData(
+      World world, Random random, int chunkX, int chunkZ, BiomeGrid biome) {
+    ChunkData chunkData = super.createChunkData(world);
+
+    // For everyblock in the chunk set the biome to plains
+    for (int x = 0; x < 16; x++) {
```

Comment:
Nitpick: we can skip this for 1.8 / sportpaper. I believe this only becomes necessary as of 1.13.

Also this method not returning null will bypass the call to `generate(`. The underlying code looks similar performance wise however.